hack/generator/pkg/functions/property_assignment_function.go

```diff
@@ -76,17 +74,17 @@ func NewPropertyAssignmentFunction(
 		idFactory:          idFactory,
 		direction:          direction,
 		conversions:        make(map[string]StoragePropertyConversion),
-		knownLocals:        astmodel.NewKnownLocalsSet(idFactory),
 		receiverName:       idFactory.CreateIdentifier(receiver.Name().Name(), astmodel.NotExported),
 		parameterName:      direction.SelectString("source", "destination"),
 	}
 
 	// Flag receiver and parameter names as used
-	result.knownLocals.Add(result.receiverName)
-	result.knownLocals.Add(result.parameterName)
+	knownLocals := astmodel.NewKnownLocalsSet(idFactory)
```

Comment:
perhaps as convenience `NewKnownLocalsSet` should also take `...string`s

Reply:
Would only have an effect in one place - but still a good idea. :+1: